### PR TITLE
Update tutorial queries and documentations

### LINKED ISSUE #187: Tutorial corpus link broken
File Not Found at 
https://mcgill-my.sharepoint.com/:f:/g/personal/michael_haaf_mcgill_ca/EjTbG6TDJOFFgAWSD6Hq1FABeakjZRkFL33z4F1DuPDcMw?e=1zQhw3

### LINKED ISSUE #191: Documentation and tutorial enhancements
Tutorial sample code and output.csv:
 - Modify queries in sample code to add ordering in the output CSV for easier comparison. 

Tutorial documentations: 
 - Code snippets: Modify code snippets to reflect changes on sample code.
 - Locations hint for Praat executable: Update Tutorials 4 and 5 to include hints on locating the Praat executable.
 - Tutorials 4/5: Encounter an EOFERROR without adding [if __name__ == '__main__':]. Update the documentation to include this.

### PR DESCRIPTION
**Pull request by lxy2304**
Closes #191 
Closes #187

### CODE REVIEW

File: docs/source/tutorial_pitch.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ to be runnable. The example given below continues to make use of the "tutorial-s
    # export_path = './results/tutorial_4_formants.csv')
    corpus_root = './data/LibriSpeech-aligned-subset/'
    corpus_name = 'tutorial-subset'
-   export_path = './results/tutorial_5_pitch.csv')
+   export_path = './results/tutorial_5_pitch.csv'
```
Comment (msonderegger) [inline docs/source/tutorial_pitch.rst:32]: good catch

File: docs/source/tutorial_formants.rst [hunk 2]
```diff
@@ -107,6 +114,8 @@ We can now query the results using a similar set of commands as in the previous
                   c.phone.F1.column_name('F1'), # the columns enriched by praat
                   c.phone.F2.column_name('F2'),
                   c.phone.F3.column_name('F3'))
+
```
Comment (msonderegger) [inline docs/source/tutorial_formants.rst:117]: add a comment somewhere (maybe in text after this code) explaining that the `order.by()` statements are not strictly necessary, but have been added to clean up the outputted file.

File: examples/tutorial/tutorial_1.py [hunk 3]
```diff
@@ -30,6 +30,7 @@
 
 with CorpusContext(corpus_name) as c:
     q = c.query_graph(c.phone).group_by(c.phone.label.column_name('phone'))
+    q = q.order_by(c.lexicon_phone.label)
```
Comment (msonderegger) [inline examples/tutorial/tutorial_1.py:33]: add a comment that `order_by` isn't strictly necessary, but will make the outputted CSV cleaner.

File: examples/tutorial/results/tutorial_1_subset_output.txt [hunk 4]
```diff
@@ -66,69 +66,69 @@ phone: Z
...
+The phone UW0 had 330 occurrences and an average duration of 0.09599999999999999.
+The phone UW1 had 6798 occurrences and an average duration of 0.09601941747572798.
+The phone UW2 had 132 occurrences and an average duration of 0.10499999999999997.
+The phone V had 10032 occurrences and an average duration of 0.0763157894736843.
+The phone W had 15774 occurrences and an average duration of 0.07301255230125502.
+The phone Y had 4686 occurrences and an average duration of 0.0891549295774647.
+The phone Z had 17226 occurrences and an average duration of 0.09003831417624542.
+The phone ZH had 132 occurrences and an average duration of 0.11999999999999962.
+The phone sil had 40062 occurrences and an average duration of 0.286013179571663.
+The phone spn had 2310 occurrences and an average duration of 0.5399999999999995.
```
[the remark this answers is not in the document]
Comment (msonderegger) [reply examples/tutorial/results/tutorial_1_subset_output.txt:134]: Is it possible your file is output after importing the full (not subset) test corpus?

File: examples/tutorial/tutorial_1_first_steps.ipynb [hunk 5]
```diff
@@ -737,84 +737,79 @@
...
+      "The phone AY2 had 792 occurrences and an average duration of 0.10999999999999983.\n",
+      "The phone B had 11154 occurrences and an average duration of 0.0695266272189347.\n",
+      "The phone CH had 3036 occurrences and an average duration of 0.11260869565217396.\n",
+      "The phone D had 26928 occurrences and an average duration of 0.06436274509803894.\n",
+      "The phone DH had 19272 occurrences and an average duration of 0.05547945205479459.\n",
+      "The phone EH0 had 330 occurrences and an average duration of 0.06400000000000004.\n",
+      "The phone EH1 had 17556 occurrences and an average duration of 0.07954887218045083.\n",
+      "The phone ER0 had 10626 occurrences and an average duration of 0.08621118012422366.\n",
+      "The phone ER1 had 2772 occurrences and an average duration of 0.12023809523809531.\n",
+      "The phone EY1 had 9240 occurrences and an average duration of 0.1429285714285711.\n",
```
Comment (msonderegger) [inline examples/tutorial/tutorial_1_first_steps.ipynb:773]: similarly to the last comment: something is wrong here, judging by the counts.

There should be 92 occurrences of `ZH`, from executing this in `LibriSpeech-aligned`:
```
grep -r "text = \"ZH" . | wc
```
This is the count in the old output, but not the new output (

File: examples/tutorial/tutorial_3.py [hunk 6]
```diff
@@ -26,6 +26,7 @@
                     c.syllable.discourse.name.column_name('file'),
                     )
     print(f"Exporting full query to {export_path}")
+    q = q.order_by(c.syllable.label)
```
Comment (msonderegger) [inline examples/tutorial/tutorial_3.py:29]: same comment, add a comment saying this is to clean up output

**Review by msonderegger — CHANGES_REQUESTED** [review]
This looks great, overall.  The issues with expected number of phones in output needs to be fixed before merging.

File: examples/tutorial/results/tutorial_1_subset_output.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (lxy2304) [reply examples/tutorial/results/tutorial_1_subset_output.txt:134]: Thanks, really good catch! The issue is that I added order_by after the group_by which is not logical as after group_by the data is already aggregated. I will make an edit.

**Review by msonderegger — APPROVED** [review]

**Pull request merged**